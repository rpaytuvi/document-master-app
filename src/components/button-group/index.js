import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {View} from 'react-native';
import {GhostButton} from '../';
import Styles from './styles';

class ButtonGroup extends Component {
  static propTypes = {
    textLeft: PropTypes.string,
    iconLeft: PropTypes.string,
    textRight: PropTypes.string,
    iconRight: PropTypes.string,
    onPressLeft: PropTypes.func.isRequired,
    onPressRight: PropTypes.func.isRequired,
    leftBigger: PropTypes.bool,
    selectedIndex: PropTypes.number,
  };

  static defaultProps = {
    textLeft: '',
    iconLeft: '',
    textRight: PropTypes.string,
    iconRight: PropTypes.string,
    onPressLeft: () => {},
    onPressRight: () => {},
    leftBigger: false,
    selectedIndex: -1,
  };

  render = () => {
    return (
      <View style={Styles.container}>
        <View
          style={
            this.props.leftBigger
              ? Styles.buttonContainerBig
              : Styles.buttonContainer
          }>
          <GhostButton
            text={this.props.textLeft}
            icon={this.props.iconLeft}
            onPress={this.props.onPressLeft}
            customStyle={Object.assign(
              {},
              Styles.leftButton,
              this.props.selectedIndex > -1 ? Styles.buttonUnselected : {},
              this.props.selectedIndex === 0 ? Styles.buttonSelected : {},
            )}
            customTextStyle={Styles.buttonText}
            customIconStyle={Object.assign(
              {},
              Styles.buttonIcon,
              this.props.selectedIndex === 0 ? Styles.buttonIconSelected : {},
            )}
          />
        </View>
        <View style={Styles.buttonContainer}>
          <GhostButton
            text={this.props.textRight}
            icon={this.props.iconRight}
            onPress={this.props.onPressRight}
            customStyle={Object.assign(
              {},
              Styles.rightButton,
              this.props.selectedIndex > -1 ? Styles.buttonUnselected : {},
              this.props.selectedIndex === 1 ? Styles.buttonSelected : {},
            )}
            customTextStyle={Styles.buttonText}
            customIconStyle={Object.assign(
              {},
              Styles.buttonIcon,
              this.props.selectedIndex === 1 ? Styles.buttonIconSelected : {},
            )}
          />
        </View>
      </View>
    );
  };
}

export default ButtonGroup;
