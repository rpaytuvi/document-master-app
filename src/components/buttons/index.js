import GhostButton from './ghost';
import HeaderButton from './header';
import MainButton from './main';

export {GhostButton, HeaderButton, MainButton};
