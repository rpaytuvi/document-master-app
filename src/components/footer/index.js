import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {View} from 'react-native';
import {MainButton} from '../';
import Styles from './styles';

class Footer extends Component {
  static propTypes = {
    buttonText: PropTypes.string.isRequired,
    buttonIcon: PropTypes.string,
    action: PropTypes.func.isRequired,
  };

  static defaultProps = {
    buttonText: '',
    buttonIcon: '',
    action: () => {},
  };

  render = () => {
    return (
      <View style={Styles.container}>
        <MainButton
          text={this.props.buttonText}
          icon={this.props.buttonIcon}
          onPress={this.props.action}
        />
      </View>
    );
  };
}

export default Footer;
