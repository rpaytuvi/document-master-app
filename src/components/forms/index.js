import DocumentForm from './document';

export {DocumentForm};
