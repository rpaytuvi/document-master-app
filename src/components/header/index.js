import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {Text, View} from 'react-native';
import {HeaderButton} from '../';
import Styles from './styles';

class Header extends Component {
  static propTypes = {
    title: PropTypes.string.isRequired,
    action: PropTypes.func.isRequired,
    withAlert: PropTypes.bool.isRequired,
  };

  static defaultProps = {
    title: '',
    action: () => {},
    withAlert: false,
  };

  render = () => {
    return (
      <View style={Styles.container}>
        <View style={Styles.titleContainer}>
          <Text style={Styles.title}>{this.props.title}</Text>
        </View>
        <View style={Styles.buttonContainer}>
          <HeaderButton
            onPress={this.props.action}
            withAlert={this.props.withAlert}
          />
        </View>
      </View>
    );
  };
}

export default Header;
