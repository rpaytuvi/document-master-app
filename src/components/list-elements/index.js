import DocumentLargeListElement from './document-large';
import DocumentSmallListElement from './document-small';

export {DocumentLargeListElement, DocumentSmallListElement};
