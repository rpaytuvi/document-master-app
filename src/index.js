import React, {Component} from 'react';
import {Provider} from 'react-redux';
import {SafeAreaProvider} from 'react-native-safe-area-context';
import {Root} from './screens';
import store from './redux';

class App extends Component {
  render = () => {
    return (
      <SafeAreaProvider>
        <Provider store={store}>
          <Root />
        </Provider>
      </SafeAreaProvider>
    );
  };
}

export default App;
