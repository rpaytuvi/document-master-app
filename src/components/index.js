import {HeaderButton, MainButton} from './buttons';
import {
  DocumentLargeListElement,
  DocumentSmallListElement,
} from './list-elements';
import {DocumentsListHeader} from './list-headers';
import {DocumentsList} from './lists';
import Footer from './footer';
import Header from './header';

export {
  HeaderButton,
  MainButton,
  DocumentLargeListElement,
  DocumentSmallListElement,
  DocumentsListHeader,
  DocumentsList,
  Footer,
  Header,
};
