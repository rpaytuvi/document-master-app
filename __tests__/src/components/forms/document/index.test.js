import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {DocumentForm} from '../../../../../src/components';

test('renders correctly', () => {
  const tree = renderer.create(<DocumentForm />).toJSON();
  expect(tree).toMatchSnapshot();
});
