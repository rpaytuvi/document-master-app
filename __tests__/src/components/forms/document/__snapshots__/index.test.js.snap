// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={false}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "flex-end",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "#e8e9ec",
          "borderRadius": 4,
          "borderTopWidth": 2,
          "flexDirection": "column",
          "margin": 32,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "padding": 16,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 24,
              "fontWeight": "600",
            }
          }
        >
          
        </Text>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderColor": "#e8e9ec",
              "borderRadius": 4,
              "borderWidth": 0,
              "flex": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "opacity": 1,
              "padding": 8,
            }
          }
        >
          <Icon
            name="close"
            size={24}
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "color": "#646d7a",
                  "marginRight": 8,
                },
                Object {
                  "marginRight": 0,
                },
              ]
            }
          />
          <Text
            style={
              Array [
                Object {
                  "color": "#646d7a",
                  "fontSize": 20,
                  "fontWeight": "600",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "padding": 16,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 16,
              "marginBottom": 8,
            }
          }
        >
          Name
        </Text>
        <TextInput
          onChangeText={[Function]}
          placeholder="Name"
          style={
            Object {
              "backgroundColor": "white",
              "borderColor": "#e8e9ec",
              "borderRadius": 4,
              "borderWidth": 2,
              "marginBottom": 16,
              "padding": 8,
            }
          }
          value=""
        />
        <Text
          style={
            Object {
              "fontSize": 16,
              "marginBottom": 8,
            }
          }
        >
          Version
        </Text>
        <TextInput
          onChangeText={[Function]}
          placeholder="Version"
          style={
            Object {
              "backgroundColor": "white",
              "borderColor": "#e8e9ec",
              "borderRadius": 4,
              "borderWidth": 2,
              "marginBottom": 16,
              "padding": 8,
            }
          }
          value=""
        />
        <Text
          style={
            Object {
              "fontSize": 16,
              "marginBottom": 8,
            }
          }
        >
          File
        </Text>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderColor": "#e8e9ec",
              "borderRadius": 4,
              "borderWidth": 2,
              "flex": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "opacity": 1,
              "padding": 8,
              "width": 150,
            }
          }
        >
          <Icon
            name="text-snippet"
            size={24}
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "color": "#646d7a",
                  "marginRight": 8,
                },
                Object {
                  "color": "#3f82f2",
                },
              ]
            }
          />
          <Text
            style={
              Array [
                Object {
                  "color": "#646d7a",
                  "fontSize": 20,
                  "fontWeight": "600",
                },
                Object {
                  "color": "#3f82f2",
                  "fontSize": 16,
                },
              ]
            }
          >
            Choose file
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#e8e9ec",
            "borderTopWidth": 2,
            "flexDirection": "row",
            "padding": 16,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#3f82f2",
              "borderRadius": 4,
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "opacity": 1,
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 20,
                "fontWeight": "600",
              }
            }
          >
            Submit
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
